h/views/account_signup.py

```diff
@@ -76,15 +81,151 @@ def validation_failure(self):
 
     @property
     def js_config(self) -> dict[str, Any]:
-        return {"csrfToken": get_csrf_token(self.request)}
+        feature_service = self.request.find_service(name="feature")
+
+        return {
+            "csrfToken": get_csrf_token(self.request),
+            "features": {
+                "log_in_with_orcid": feature_service.enabled(
+                    "log_in_with_orcid", user=None
+                )
+            },
+        }
+
+
+class AuthJWTDecodeError(Exception):
+    """Decoding the `auth` JWT query param failed."""
+
+
+@dataclass
+class IDInfo:
+    sub: str
```

Comment:
This is the dataclass used for the JWT query param that our `/oidc/redirect/orcid` view uses to pass the user's ORCID iD to our `/signup/orcid` view when redirecting the browser.

I'm using the standard JWT `sub` (subject) claim for the ORCID iD (and in future this will be reused for Google and Facebook IDs). The JWT spec says that `sub`'s should be unique within a given `aud` and indeed this will be (since we'll be using a different `aud` for each account provider).